app/services/unwithdraw_service.rb

```diff
@@ -0,0 +1,32 @@
+# frozen_string_literal: true
+
+class UnwithdrawService
+  def process(edition, user = nil)
+    Document.transaction do
+      edition.document.lock!
```

Comment:
We don't really need to do this lock as it's not called from a rake task like the unpublishing service is, I'd remove this until we have a secondary use for this class.